platform/parsers/elastic_query_dsl/pancake_aggregation_parser_buckets.go

```diff
@@ -178,18 +184,24 @@ func (cw *ClickhouseQueryTranslator) parseTermsAggregation(aggregation *pancakeA
 		aggregation.filterOutEmptyKeyBucket = true
 	}
 
-	const defaultSize = 10
-	size := cw.parseSize(params, defaultSize)
-
 	orderBy, err := cw.parseOrder(params, []model.Expr{field})
 	if err != nil {
 		return err
 	}
 
 	aggregation.queryType = terms
 	aggregation.selectedColumns = append(aggregation.selectedColumns, field)
-	aggregation.limit = size
+	aggregation.limit = cw.parseSize(params, defaultSize)
 	aggregation.orderBy = orderBy
+	if minDocCount > 1 {
+		// need to filter out buckets with doc_count < min_doc_count, so first order by is (count() >= min_doc_count) DESC
+		condition := model.NewInfixExpr(model.NewCountFunc(), ">=", model.NewLiteral(minDocCount))
```

Comment:
I don't understand why we do that here. Probably would love to talk about this logic. Previously similar filtering was done on client side and that kind of logic was not neccessary.

Reply:
Ok, please add just better comment with example.